### PR TITLE
Allow replace operation on empty default block in Zoom Out

### PR DESCRIPTION
**Pull request by getdave**
## What?

Allows for drag and drop of Patterns in Zoom Out if the only block is an unmodified default block (normally `core/paragraph`).

Fixes part of https://github.com/WordPress/gutenberg/issues/67564

## Why?

Currently on `trunk` when editing a Page in the Site Editor you can end up with a placeholder `core/paragraph` block in witin the `core/post-content` block.

When in Zoom Out you drag and drop a Pattern onto this block there is no drop zone rendered.

This PR fixes that to show the drop zone. It also works if there is an unmodified default block _anywhere_ within the block list which can happen if you click a pattern and insert into an empty page as the default block remains.

## How?

Allows for a `replace` operation to be handled in Zoom Out but _only_ if the block being replaced is an unmodified default block (i.e. `core/paragraph`).

This in turn allows the zoom out separator to be rendered.

## Testing Instructions

- Site Editor
- New Page
- Empty all content
- See that there is a default paragraph
- Open Zoom Out
- Drag and drop a Pattern and see the drop zone rendered.
- Undo (back to default para only)
- _Click_ any Pattern to insert into the Page. You should see the default paragraph remains (this is not a bug but intended functionality).
- Now attempt drag and drop of Pattern again above/below newly inserted Pattern checking that drop zones are rendered.

## Screenshots or screencast 

https://github.com/user-attachments/assets/f0a904a8-9ab6-4234-a63d-a95bff74e460

### CODE REVIEW

**Comment by getdave** [comment]
Noting that I think we need to follow up with this to:

- _replace_ the unmodified default block when clicking to insert Patterns in Zoom Out.
- also allow for dropping onto a completely empty block list (one that doesn't even contain a empty default block).

File: packages/block-editor/src/components/use-block-drop-zone/index.js [hunk 1]
```diff
@@ -456,7 +456,15 @@ export default function useBlockDropZone( {
 				const [ targetIndex, operation, nearestSide ] =
 					dropTargetPosition;
 
-				if ( isZoomOut() && operation !== 'insert' ) {
+				const isTargetIndexEmptyDefaultBlock =
+					operation === 'replace' &&
+					blocksData[ targetIndex ]?.isUnmodifiedDefaultBlock;
```
Comment (draganescu) [inline packages/block-editor/src/components/use-block-drop-zone/index.js:460]: It's a bit confusing to read operation === 'replace' and later operation !== 'insert' and also why does the kind of operation determine if isTargetIndexEmptyDefaultBlock, as it feels the isUnmodifiedDefaultBlock flag should be enough.

Comment (getdave) [reply packages/block-editor/src/components/use-block-drop-zone/index.js:460]: You know what I think we might be able to remove the `operation` check like you suggest.

Comment (getdave) [reply packages/block-editor/src/components/use-block-drop-zone/index.js:460]: ✅ Does it work for you?

**Review by draganescu — APPROVED** [review]

**Pull request merged**